src/machine/machine_atmega.go

```diff
@@ -5,11 +5,95 @@ package machine
 
 import (
 	"device/avr"
+	"errors"
 	"runtime/interrupt"
 	"runtime/volatile"
 	"unsafe"
 )
 
+// EEPROM on ATMega
+type EEPROM struct {
+}
+
+var EEPROM0 = EEPROM{}
+
+// setAddress sets the address for a given read or write into the MCUs EEARH/L register.
+func (e EEPROM) setAddress(addr int16) error {
+	if addr < 0 || addr > int16(e.Size()) {
```

Comment:
I think this is an off-by-one.

```suggestion
	if addr < 0 || addr >= int16(e.Size()) {
```